overlord/snapstate/storehelpers.go

```diff
@@ -160,6 +160,7 @@ func updateInfo(st *state.State, snapst *SnapState, opts *updateInfoOpts, userID
 		Action:       "refresh",
 		InstanceName: curInfo.InstanceName(),
 		SnapID:       curInfo.SnapID,
+		Epoch:        &curInfo.Epoch,
```

Comment:
I think we should use the full expressivity of the API and use nil here, accept for the amend case

Reply:
the epoch in the action overrides the epoch in context, even if it's `null`. We've got two options: send the full epoch, or not send the epoch at all.

For install, we need to send a `null` epoch so the store knows snapd knows about epochs.

Reply:
yes, this is correct and I was confused,  we discussed further and given that we don't have a use case to send "epoch" in "refresh" actions we have a plan forward